### PR TITLE
October update to devx_roadmap

### PR DESCRIPTION
**Pull request by srinath-mysten**

### CODE REVIEW

File: DEVX_ROADMAP.md
Comment (janetwu11) [inline DEVX_ROADMAP.md:12]: -I think we can omit metrics, logging,and testing because these are are not visible to developers.
- should add 'Typescript' in front of migration guide to be specific.

File: DEVX_ROADMAP.md
Comment (janetwu11) [inline DEVX_ROADMAP.md:12]: Since you used the 'breaking change' notation for other RPC calls later, do you also want to apply that to Gateway deprecationw hich is breaking in that the RPC endpoint is going away?

File: DEVX_ROADMAP.md
Comment (janetwu11) [inline DEVX_ROADMAP.md:8]: Should we call this 'dynamic' child object to be consistent with the naming later in the Move section?

File: DEVX_ROADMAP.md
Comment (janetwu11) [inline DEVX_ROADMAP.md:38]: Should we link to Chris' write up here for folks looking for further information? https://www.notion.so/mystenlabs/RFC-Faucet-Improvement-Plan-f706cd7411c046bc8fca3f1a7a8b559b

**Review by janetwu11 — COMMENTED** [review]
Added some inline comments.

File: DEVX_ROADMAP.md
Comment (sblackshear) [inline DEVX_ROADMAP.md:16]:
```suggestion
* Adding `PaySui` transaction type, which takes multiple coins and transfer them to a single recipient, while paying gas with the input coins. Deprecating `TransferSui` as `PaySui` is a generalized version of it [**Breaking Change**].
```

File: DEVX_ROADMAP.md
Comment (sblackshear) [inline DEVX_ROADMAP.md:9]: Perhaps worth linking to the NFT standard draft here as well: https://github.com/MystenLabs/sui/pull/4887

File: DEVX_ROADMAP.md
Comment (sblackshear) [inline DEVX_ROADMAP.md:32]: Other items to mention
- Shared coins via `Safe` module: https://github.com/MystenLabs/sui/pull/4680
- Support for type reflection: https://github.com/move-language/move/pull/566
- Support for `_`'s in integer literals (e.g., `let one_million = 1_000_000`)

File: DEVX_ROADMAP.md
Comment (sblackshear) [inline DEVX_ROADMAP.md:39]: - Source verification for dependent packages enforced at publish time

File: DEVX_ROADMAP.md
Comment (srinath-mysten) [reply DEVX_ROADMAP.md:43]: We don't have an RFC in github for this yet.

**Review by sblackshear — APPROVED** [review]

File: DEVX_ROADMAP.md
Comment (666lcz) [inline DEVX_ROADMAP.md:43]:
```suggestion
* As a precursor to integrating Faucet into Sui CLI/SDKs/Wallet, rate-limiting will be enforced on the faucet server to no more than 20 requests in 24 hours, originating from the same ip address.
```

Based on[ internal discussion](https://mysten-labs.slack.com/archives/C03GKUEA5PF/p1665009940561759?thread_ts=1665008605.936689&cid=C03GKUEA5PF), we will just keep the latter for now

**Comment by srinath-mysten** [comment]
I'll wait for one more approval before merging.

File: DEVX_ROADMAP.md
Comment (janetwu11) [inline DEVX_ROADMAP.md:2]: we are now 10/12

**Review by janetwu11 — COMMENTED** [review]
added a couple more small things but otherwise GTG.

File: DEVX_ROADMAP.md
Comment (randall-Mysten) [reply DEVX_ROADMAP.md:31]: or Devnet, but not devNet. Thank you!

**Pull request merged**
